### ISSUE TITLE
[Suggestion] "CronManager" with easytimer.js

### ISSUE THREAD

**Issue by exodus4d** [albert-gonzalez/easytimer.js#90]
@albert-gonzalez awesome project! - Not sure if this proposal will make it into the projects scope some day. Feel free to close this issue if it is not relevant.

### Issue:
If you have a _SPA_ where multiple _EasyTimer_ instances are initialized dynamically to handle various tasks, the browser might run into performance issues to handle all the _EasyTimer_ interval functions...

### Suggestion:
I made a (vanilla JS/ES6 compatible) [_CronManager_](https://github.com/exodus4d/pathfinder/blob/develop/js/app/lib/cron.js#L185) class, that manages a **single** _EasyTimer_ instance and handles _attached_ all [_Task_](https://github.com/exodus4d/pathfinder/blob/develop/js/app/lib/cron.js#L53)´s dynamically.
The concept behind this is to have a **single** _high frequency EasyTimer_ running in the background and _Task_ execution is dispatched by the _CronManager_.

### Examples:
```js
// global instance used across the app
let cron = new CronManager();

// 1. task runs every second ------------------------------------------------------------
let task1 = cron.new('task1', {precision: 'seconds'});
task1.task = timer => {
    console.log('task1', timer.getTotalTimeValues());
};
cron.set(task1);

// 2. task runs every 2nd minute --------------------------------------------------------
let task2 = cron.new('task2', {precision: 'minutes', interval: 2});
task2.task = () => console.log('task2');
cron.set(task2);

// 3. task runs only once at dueDate ----------------------------------------------------
let dueDate = new Date();
dueDate.setSeconds(dueDate.getSeconds() + 5);

let task3 = cron.new('task3', {precision: 'seconds', dueDate: dueDate});
task3.task = () => console.log('task3');
cron.set(task3);

// 4. timeout (100ms) for long running tasks - Promise.race() --------------------------
let task4 = cron.new('task4', {precision: 'seconds', timeout: 100});
task4.task = () => console.log('task4 long execution time....');
cron.set(task4);

// 5. Promis() chain resolver task -----------------------------------------------------
let task5 = cron.new('task5', {precision: 'hours', timeout: 150});
task5.task = (timer, task) => {
    return new Promise((resolve, reject) => {
        resolve('task5 Promise1');
    }).then(payload => new Promise((resolve, reject) => {
        resolve('task5 Promise2');
    });
};
cron.set(task5);

// delete tasks if they are no longer needed ===========================================
cron.delete('task1');
cron.delete('task4');
cron.clear(); // remove all 
```

### More features:
- _EasyTimer_ is auto `start()`/`stop()` if tasks added
- Run tasks parallel (if prev. run has not finished yet) - no skip
  ```js 
  let task1 = cron.new('task1', {isParallel: true});
  ```
- Change task config dynamically 
  - Increase interval by 1s after each run
  ```js
  let task1 = cron.get('task1');
  task1.task = (timer, task) => {
      // do something...
      task.set('interval', task.get('interval') + 1)
  };
  ```
  - Remove task after 3 runs
  ```js
  let task2 = cron.get('task2');
  task2.task = (timer, task) => {
      // do something...
      if (task.runCount ==  3) {
          task.delete();
      }
  };
  ```

**Comment by albert-gonzalez** [comment]
Hi!

Thanks a lot for your suggestion! I think it's out of the project's scope because I prefer to keep this library as a simple timer. Your CronManager is pretty awesome though! Maybe you can publish it as a package.

I close this issue for now, but if you need anything, please let me know!

Thanks again!
